cpp/src/arrow/compute/function.h

```diff
@@ -159,6 +159,20 @@ struct ARROW_EXPORT FunctionDoc {
   static const FunctionDoc& Empty();
 };
 
+/// \brief An executor of a function with a preconfigured kernel
+struct ARROW_EXPORT FunctionExecutor {
```

Comment:
Style nit, but we should probably use `struct` for data classes, not interfaces like this.
```suggestion
class ARROW_EXPORT FunctionExecutor {
```